core/src/test/java/org/apache/calcite/rex/RexExecutorTest.java

```diff
@@ -175,6 +176,58 @@ private void checkConstant(final Object operand,
     });
   }
 
+  @Test public void testUserFromContext() throws Exception {
+    testContextLiteral(SqlStdOperatorTable.USER,
+                       DataContext.Variable.USER, "happyCalciteUser");
+  }
+
+  @Test public void testSystemUserFromContext() throws Exception {
+    testContextLiteral(SqlStdOperatorTable.SYSTEM_USER,
+                       DataContext.Variable.SYSTEM_USER, "");
+  }
+
+  @Test public void testTimestampFromContext() throws Exception {
+    // current timestamp currently rounds to the nearest second (?)
+    long val = System.currentTimeMillis() / 1000 * 1000;
+    testContextLiteral(SqlStdOperatorTable.CURRENT_TIMESTAMP,
+                       DataContext.Variable.CURRENT_TIMESTAMP, val);
+  }
+
+  /**
+   * Ensure that for a given context operator, the correct value is retrieved from the DataContext.
+   * @param operator The Operator to check
+   * @param variable The DataContext variable this operator should be bound to.
+   * @param value The expected value to retrieve.
+   */
+  private void testContextLiteral(
+          final SqlOperator operator,
+          final DataContext.Variable variable,
+          final Object value) {
+    Frameworks.withPrepare(new Frameworks.PrepareAction<Void>() {
+      public Void apply(final RelOptCluster cluster,
+                        final RelOptSchema relOptSchema,
+                        final SchemaPlus rootSchema,
+                        final CalciteServerStatement statement) {
+        final RexBuilder rexBuilder = cluster.getRexBuilder();
+        final RexExecutorImpl executor = new RexExecutorImpl(
+                new SingleValueDataContext(variable.camelName, value));
+        try {
+          checkConstant(value, new Function<RexBuilder, RexNode>() {
+            public RexNode apply(RexBuilder builder) {
+              List<RexNode> output = new ArrayList<>();
+              executor.reduce(rexBuilder,
+                              ImmutableList.of(rexBuilder.makeCall(operator)), output);
+              return output.get(0);
+            }
+          });
+        } catch (Exception e) {
+          throw new RuntimeException(e);
```

Comment:
Just a minor note: `throw org.apache.calcite.util.TestUtil#rethrow(java.lang.Throwable)` should be used to rethrow exceptions in the test code.

Reply:
Thanks @vlsi. I have made the change.

Reply:
@vlsi Is it possible to add a forbidden-apis check to flag these violations?

Reply:
@julianhyde , I've no idea. It could probably work if we split current execution in two:

```
<execution>
            <goals>
              <goal>check</goal>
              <goal>testCheck</goal>
            </goals>
</execution>
```

However I find replacing Apache JMeter's Ant-based build with a Gradle-based one to be a more fascinating thing a the moment.

PS. The more interesting thing here is `new Function` which should be expressed as a lambda to reduce verbosity. 

PPS. I do like the way Gradle feels. It just automatically rebuilds the required modules. For instance, in Maven you need to "mvn install" the core before you run "csv tests", however Gradle does that transparently, and it can even do that across projects. It is really amazing.